crypto/context.c

```diff
@@ -53,20 +58,32 @@ static int context_init(OPENSSL_CTX *ctx)
     if (ctx->oncelock == NULL)
         goto err;
 
-    for (i = 0; i < OPENSSL_CTX_MAX_INDEXES; i++)
+    for (i = 0; i < OPENSSL_CTX_MAX_INDEXES; i++) {
+        ctx->index_locks[i] = CRYPTO_THREAD_lock_new();
         ctx->dyn_indexes[i] = -1;
+        if (ctx->index_locks[i] == NULL)
+            goto err;
+    }
 
+    /* OPENSSL_CTX is built on top of ex_data so we initialise that directly */
     if (!do_ex_data_init(ctx))
         goto err;
+    exdata_done = 1;
 
     if (!crypto_new_ex_data_ex(ctx, CRYPTO_EX_INDEX_OPENSSL_CTX, NULL,
                                &ctx->data)) {
         crypto_cleanup_all_ex_data_int(ctx);
         goto err;
     }
 
+    /* Everything depends on properties, so we also pre-initialise that */
+    if (!ossl_property_parse_init(ctx))
```

Comment:
I think that `do_method_store_init` might be a better initialisation call under a `openssl_ctx_run_once` cover.

Reply:
`do_method_store_init()` is in `crypto/property/property.c`.  It's the RUN_ONCE that calls `ossl_property_parse_init()` (and that's the only thing it does).

The `ossl_property_parse_init()` function can safely be called multiple times, it is preloading our property names and the Boolean values.  If any already exist, they aren't added repeatedly.

So it would be safe to leave this as it is (and perhaps remove `do_method_store_init()` completely).